src/lib/device/device_manager.cpp

```diff
@@ -176,6 +177,27 @@ void DeviceManager::load_devices() {
     load_vdevs();
 }
 
+void DeviceManager::commit_formatting() {
+    auto buf = hs_utils::iobuf_alloc(hs_super_blk::first_block_size(), sisl::buftag::superblk, 512);
+    for (auto& pdev : m_all_pdevs) {
+        if (!pdev) { continue; }
+
+        auto err = pdev->read_super_block(buf, hs_super_blk::first_block_size(), hs_super_blk::first_block_offset());
+        if (err) {
+            LOGERROR("Failed to read first block from device={}, error={}", pdev->get_devname(), err.message());
+            continue;
+        }
+
+        first_block* fblk = r_cast< first_block* >(buf);
+        fblk->formatting_done = 0x1;
```

Comment:
In the case of there are three pdevs, and 1 got persisted with formatting_done set to 0x1, before 2 other pdevs persist their superblock, we crash. 

On reboot, what we want is we should return need_format true in homestore::start(). 
However, on reboot in  above case, in `DeviceManager::DeviceManager`, when we detect 1st pdev is_valid() is true (its formatting_done already set to 0x1), we will set `m_first_time_boot` as false, which will return `need_format` as false, which is not expected, right? 

We want to change `DeviceManager::DeviceManager` to only return m_first_time_boot as true when all pdevs is_valid() return true, right?

Reply:
This needs to be fixed globally and decide strategically, not just for formatting, but in general its a valid device or not. Currently we need superblk from only one device to be valid and if they are, we proceed. We can choose to do it only if all (but we need to worry about new device insertion). I would just use is_valid() and let the DeviceManager::DeviceManager make a decision.